crates/optimism/flashblocks/src/service.rs

```diff
@@ -37,13 +40,16 @@ pub struct FlashBlockService<
     blocks: Vec<FlashBlock>,
     evm_config: EvmConfig,
     provider: Provider,
+    canon_receiver: CanonStateNotifications<N>,
+    tip: Option<BlockHash>,
```

Comment:
we can't rely on this alone because this still does not guarantee that:

this is the latest is parent-hash:

https://github.com/paradigmxyz/reth/blob/reth-optimism-flashblocks11/crates/optimism/flashblocks/src/service.rs#L122-L133

so we dont need to track this here